source/type/type.cpp

```diff
@@ -176,26 +176,48 @@ bool StopTime::is_valid_day(u_int32_t day, const bool is_arrival, const RTLevel
 }
 
 const StopTime* StopTime::get_base_stop_time() const {
-    const nt::StopTime* st_base = nullptr;
-    size_t nb_st_found = 0;
-    if (vehicle_journey == nullptr) { return nullptr; }
+    
+    if (vehicle_journey == nullptr) { 
+        return nullptr; 
+    }
+
     auto base_vj = vehicle_journey->get_corresponding_base();
-    if (base_vj == nullptr) { return nullptr; }
+    
+    if (base_vj == nullptr) { 
+        return nullptr; 
+    }
 
-    if (vehicle_journey == base_vj) { return this; }
+    if (vehicle_journey == base_vj) { 
+        return this; 
+    }
+    
+    std::vector<const nt::StopTime*> st_base;
     for (const auto& st_it: base_vj->stop_time_list) {
         if (stop_point == st_it.stop_point) {
-            st_base = &st_it;
-            ++nb_st_found;
+            st_base.push_back(&st_it);
         }
     }
-    //TODO handle lolipop vj, bob-commerce-commerce-bobito vj...
-    if (nb_st_found != 1) {
+
+    if (st_base.empty()) { return nullptr; }
+
+    // handle lolipop 
+    size_t nb_rt_st_found = 0;
+    if (st_base.size() > 1) {
+        for (const auto& st : vehicle_journey->stop_time_list) {
+            if (&st == this) {
+                return st_base[nb_rt_st_found];
```

Comment:
:boom: if realtime_VJ passes more times by given StopPoint than base_VJ.

Reply:
And, if you first search for the current vj to have the rank, and then search in the base, you don't need to allocate a vector.